app/src/main/java/com/kickstarter/viewmodels/ProjectNotificationViewModel.java

```diff
@@ -20,80 +17,101 @@
 import static com.kickstarter.libs.rx.transformers.Transformers.takePairWhen;
 import static com.kickstarter.libs.rx.transformers.Transformers.values;
 
-public class ProjectNotificationViewModel extends ActivityViewModel<ProjectNotificationViewHolder> implements
-  ProjectNotificationViewModelInputs, ProjectNotificationViewModelOutputs, ProjectNotificationViewModelErrors {
-  public ProjectNotificationViewModel(final @NonNull Environment environment) {
-    super(environment);
-
-    final ApiClientType client = environment.apiClient();
-
-    // When the enable switch is clicked, update the project notification.
-    final Observable<Notification<ProjectNotification>> updateNotification = this.projectNotification
-      .compose(takePairWhen(this.enabledSwitchClick))
-      .switchMap(ne ->
-        client
-          .updateProjectNotifications(ne.first, ne.second)
-          .materialize()
-      )
-      .share();
-
-    updateNotification
-      .compose(values())
-      .compose(bindToLifecycle())
-      .subscribe(this.projectNotification::onNext);
-
-    updateNotification
-      .compose(errors())
-      .compose(bindToLifecycle())
-      .subscribe(__ -> this.showUnableToSaveProjectNotificationError.onNext(null));
-
-    // Update the project name when a project notification emits.
-    this.projectNotification
-      .map(n -> n.project().name())
-      .compose(bindToLifecycle())
-      .subscribe(this.projectName::onNext);
-
-    // Update the enabled switch when a project notification emits.
-    this.projectNotification
-      .map(n -> n.email() && n.mobile())
-      .compose(bindToLifecycle())
-      .subscribe(this.enabledSwitch::onNext);
+public interface ProjectNotificationViewModel {
+  interface Inputs {
+    /**
+     * Call when the enable switch is clicked.
+     */
```

Comment:
would be nice to reformat these javadoc comments to `/** content */` to make the VM a lil more readable